plugin/pkg/admission/limitranger/admission.go

```diff
@@ -551,14 +552,24 @@ func PodValidateLimitFunc(limitRange *corev1.LimitRange, pod *api.Pod) error {
 type podResourcesOptions struct {
 	// InPlacePodVerticalScalingEnabled indicates that the in-place pod vertical scaling feature gate is enabled.
 	InPlacePodVerticalScalingEnabled bool
+	// PodLevelResourcesEnabled indicates that the PodLevelResources feature gate is
+	// enabled.
+	PodLevelResourcesEnabled bool
 }
 
 // podRequests is a simplified version of pkg/api/v1/resource/PodRequests that operates against the core version of
 // pod. Any changes to that calculation should be reflected here.
 // TODO: Maybe we can consider doing a partial conversion of the pod to a v1
 // type and then using the pkg/api/v1/resource/PodRequests.
+// TODO(ndixita): PodRequests method exists in
+// staging/src/k8s.io/component-helpers/resource/helpers.go. Refactor the code to
+// avoid duplicating podRequests method.
 func podRequests(pod *api.Pod, opts podResourcesOptions) api.ResourceList {
 	reqs := api.ResourceList{}
+	if opts.PodLevelResourcesEnabled && pod.Spec.Resources != nil {
+		addResourceList(reqs, pod.Spec.Resources.Requests)
```

Comment:
So I guess this is because podspec resource is all we need to care about, e.g. we have considered the initContainer in the api server, so we can directly return here, right? Will the pod level resources be partly empty, like requests are set but limits are empty?

Reply:
I modified the logic since I changed the same in the staging/src/k8s.io/component-helpers/resource/helpers.go. Podspec resoure is all we need to care about only for cpu and memory resources since only those are supported at pod-level.